diff -Nuard hal-0.4.2.orig/fdi/90defaultpolicy/storage-policy.fdi hal-0.4.2/fdi/90defaultpolicy/storage-policy.fdi
--- hal-0.4.2.orig/fdi/90defaultpolicy/storage-policy.fdi	2004-12-10 17:37:47.125532872 +0100
+++ hal-0.4.2/fdi/90defaultpolicy/storage-policy.fdi	2004-12-10 17:38:04.173941120 +0100
@@ -6,12 +6,13 @@
   <device>
     <match key="info.udi" string="/org/freedesktop/Hal/devices/computer">
       <merge key="storage.policy.default.mount_root" type="string">/media</merge>
-      <merge key="storage.policy.default.use_managed_keyword" type="bool">true</merge>
+      <merge key="storage.policy.default.use_managed_keyword" type="bool">false</merge>
       <merge key="storage.policy.default.managed_keyword.primary" type="string">managed</merge>
       <merge key="storage.policy.default.managed_keyword.secondary" type="string">kudzu</merge>
       <merge key="storage.policy.default.mount_option.noauto" type="bool">true</merge>
-      <merge key="storage.policy.default.mount_option.pamconsole" type="bool">true</merge>
+      <merge key="storage.policy.default.mount_option.pamconsole" type="bool">false</merge>
       <merge key="storage.policy.default.mount_option.exec" type="bool">true</merge>
+      <merge key="storage.policy.default.mount_option.users" type="bool">true</merge>
     </match>
   </device>
 
