Index: fdi/90defaultpolicy/storage-policy.fdi
===================================================================
RCS file: /cvs/hal/hal/fdi/90defaultpolicy/storage-policy.fdi,v
retrieving revision 1.6
diff -u -r1.6 storage-policy.fdi
--- fdi/90defaultpolicy/storage-policy.fdi	18 Oct 2004 23:54:11 -0000	1.6
+++ fdi/90defaultpolicy/storage-policy.fdi	25 Oct 2004 16:25:45 -0000
@@ -39,7 +39,7 @@
     <match key="storage.no_partitions_hint" bool="true">
       <!-- optical drives -->
       <match key="storage.drive_type" string="cdrom">
-	<merge key="storage.policy.mount_filesystem" type="string">iso9660,udf</merge>
+	<merge key="storage.policy.mount_filesystem" type="string">auto</merge>
 	<merge key="storage.policy.desired_mount_point" type="string">cdrom</merge>
 	<merge key="storage.policy.mount_option.ro" type="bool">true</merge>
 	<match key="storage.cdrom.cdr" bool="true">
