--- hal-0.5.10/configure.in.orig	2007-10-31 20:02:08.847844000 +0100
+++ hal-0.5.10/configure.in	2007-10-31 20:20:27.462450588 +0100
@@ -242,8 +242,8 @@
 if test "x$use_parted" = "xyes" ; then
   USE_PARTED=yes
   AC_DEFINE(USE_PARTED,1,[Whether libparted is to be used])
-  AC_CHECK_LIB(uuid, uuid_generate, [], AC_MSG_ERROR([*** uuid library (libuuid) not found]))
-  AC_CHECK_LIB(dl, dlopen,          [], AC_MSG_ERROR([*** dl library (libdl) not found]))
+  AC_CHECK_LIB(uuid, uuid_generate, [:], AC_MSG_ERROR([*** uuid library (libuuid) not found]))
+  AC_CHECK_LIB(dl, dlopen,          [:], AC_MSG_ERROR([*** dl library (libdl) not found]))
 
   AC_MSG_CHECKING([for libparted == 1.7.1 or >= 1.8.0])
   if ! $PKG_CONFIG --atleast-version 1.8.0 libparted; then
