--- hal-0.5.10/hald/linux/addons/addon-macbook-backlight.c.dbus 2007-12-06 17:56:21.000000000 +0100
+++ hal-0.5.10/hald/linux/addons/addon-macbook-backlight.c 2007-12-06 17:57:01.000000000 +0100
@@ -320,6 +320,7 @@ main (int argc, char **argv)
 
 	conn = libhal_ctx_get_dbus_connection (halctx);
 	dbus_connection_setup_with_g_main (conn, NULL);
+	dbus_connection_set_exit_on_disconnect (conn, 0);
 
 	dbus_connection_add_filter (conn, filter_function, NULL, NULL);
 
--- hal-0.5.10/hald/linux/addons/addon-macbookpro-backlight.c.dbus 2007-12-06 17:56:26.000000000 +0100
+++ hal-0.5.10/hald/linux/addons/addon-macbookpro-backlight.c 2007-12-06 17:57:37.000000000 +0100
@@ -502,6 +502,7 @@ main (int argc, char *argv[])
 
 	conn = libhal_ctx_get_dbus_connection (halctx);
 	dbus_connection_setup_with_g_main (conn, NULL);
+	dbus_connection_set_exit_on_disconnect (conn, 0);
 
 	dbus_connection_add_filter (conn, filter_function, NULL, NULL);
 
--- hal-0.5.10/hald/linux/addons/addon-storage.c.dbus 2007-12-06 17:56:33.000000000 +0100
+++ hal-0.5.10/hald/linux/addons/addon-storage.c 2007-12-06 17:58:43.000000000 +0100
@@ -657,6 +657,7 @@ main (int argc, char *argv[])
 	}
 	loop = g_main_loop_new (NULL, FALSE);
 	dbus_connection_setup_with_g_main (con, NULL);
+	dbus_connection_set_exit_on_disconnect (con, 0);
 
 	dbus_error_init (&error);
 	if ((ctx = libhal_ctx_init_direct (&error)) == NULL) 
