Index: tools/hal-device.c
===================================================================
RCS file: /cvs/hal/hal/tools/hal-device.c,v
retrieving revision 1.5
diff -u -p -r1.5 hal-device.c
--- tools/hal-device.c	12 May 2006 14:34:28 -0000	1.5
+++ tools/hal-device.c	12 Jul 2006 10:49:02 -0000
@@ -155,7 +155,7 @@ int main(int argc, char **argv)
 
 	libhal_ctx_shutdown(hal_ctx, &error);
 	libhal_ctx_free(hal_ctx);
-	dbus_connection_disconnect(conn);
+	dbus_connection_close(conn);
 	dbus_connection_unref(conn);
 	dbus_error_free(&error);
 
Index: tools/lshal.c
===================================================================
RCS file: /cvs/hal/hal/tools/lshal.c,v
retrieving revision 1.20
diff -u -p -r1.20 lshal.c
--- tools/lshal.c	5 Mar 2006 00:41:50 -0000	1.20
+++ tools/lshal.c	12 Jul 2006 10:49:02 -0000
@@ -705,7 +705,7 @@ main (int argc, char *argv[])
 	libhal_ctx_shutdown (hal_ctx, &error);
 	libhal_ctx_free (hal_ctx);
 
-	dbus_connection_disconnect (conn);
+	dbus_connection_close (conn);
 	dbus_connection_unref (conn);
 
 	if (show_device)
