diff -Nur hal-0.5.3.orig/tools/device-manager/DeviceManager.py hal-0.5.3/tools/device-manager/DeviceManager.py
--- hal-0.5.3.orig/tools/device-manager/DeviceManager.py	2005-07-11 18:26:58.000000000 +0100
+++ hal-0.5.3/tools/device-manager/DeviceManager.py	2005-07-13 19:16:10.000000000 +0100
@@ -264,16 +264,19 @@
         for name in device_names:
             device_dbus_obj = self.bus.get_object("org.freedesktop.Hal" ,name)
             properties = device_dbus_obj.GetAllProperties(dbus_interface="org.freedesktop.Hal.Device")
-            try:
-                parent_name = properties["info.parent"]
-            except KeyError:
-                # no parent, must be parent of virtual_root
-                parent_name = "/"
-            except TypeError:
-                print "Error: no properties for device %s"%name
-                continue
-            device = Device(name, parent_name, properties)
-            self.device_list.append(device)
+            if properties == None:
+                print name + " has no properties"
+            else:
+                try:
+                    parent_name = properties["info.parent"]
+                except KeyError:
+                    # no parent, must be parent of virtual_root
+                    parent_name = "/"
+                except TypeError:
+                    print "Error: no properties for device %s"%name
+                    continue
+                device = Device(name, parent_name, properties)
+                self.device_list.append(device)
 
         # set parent_device and children for each Device object
         for device in self.device_list:
