--- hal-0.5.12/hald/linux/probing/probe-volume.c.orig	2009-05-10 20:11:44.000000000 +0100
+++ hal-0.5.12/hald/linux/probing/probe-volume.c	2009-05-24 09:47:11.000000000 +0100
@@ -97,7 +97,7 @@
 
 	if (blkid_probe_lookup_value(pr, "TYPE", &type, NULL))
 		type = "";
-	if (libhal_changeset_set_property_string (cs, "volume.fstype", type))
+	if (!libhal_changeset_set_property_string (cs, "volume.fstype", type))
 		libhal_changeset_set_property_string (cs, "volume.fstype", "");
 	HAL_DEBUG(("volume.fstype = '%s'", type));
 
--- hal-0.5.12/tools/hal-storage-mount.c.orig	2009-05-10 20:11:44.000000000 +0100
+++ hal-0.5.12/tools/hal-storage-mount.c	2009-05-24 09:36:51.000000000 +0100
@@ -786,7 +786,8 @@
 		/* don't consider uid= on vfat, iso9660, hfs and udf change-uid for the purpose of policy
 		 * (since these doesn't contain uid/gid bits) 
 		 */
-		if (strcmp (libhal_volume_get_fstype (volume), "vfat") != 0 &&
+		if (libhal_volume_get_fstype (volume) != NULL &&
+		    strcmp (libhal_volume_get_fstype (volume), "vfat") != 0 &&
 		    strcmp (libhal_volume_get_fstype (volume), "ntfs") != 0 &&
 		    strcmp (libhal_volume_get_fstype (volume), "ntfs-3g") != 0 &&
 		    strcmp (libhal_volume_get_fstype (volume), "iso9660") != 0 &&
