diff -urN hal-0.5.10/tools/linux/hal-system-power-hibernate-linux hal-0.5.10.new/tools/linux/hal-system-power-hibernate-linux
--- hal-0.5.10/tools/linux/hal-system-power-hibernate-linux	2007-07-26 20:00:28.000000000 +0200
+++ hal-0.5.10.new/tools/linux/hal-system-power-hibernate-linux	2007-10-14 00:20:17.000000000 +0200
@@ -23,9 +23,26 @@
 [ "$HAL_PROP_POWER_MANAGEMENT_QUIRK_NONE" = "true" ] && QUIRKS="$QUIRKS --quirk-none"
 
 # We only support pm-utils
-if [ -x /usr/sbin/pm-hibernate ] ; then
+if [ -x "/usr/sbin/pm-hibernate" ] ; then
 	/usr/sbin/pm-hibernate $QUIRKS
 	RET=$?
+elif [ -x "/usr/sbin/s2disk" ] ; then
+	/usr/sbin/s2disk
+	RET=$?
+elif [ -x "/usr/sbin/hibernate" ] ; then
+	# Suspend2 tools installed
+	/usr/sbin/hibernate --force
+	RET=$?
+elif [ -x "/usr/bin/powersave" ] ; then
+	dbus-send --system --dest=com.novell.powersave --print-reply /com/novell/powersave com.novell.powersave.action.SuspendToDisk
+	RET=$?
+elif [ -x "/usr/sbin/pmsuspend" ] ; then
+	/usr/sbin/pmsuspend disk
+	RET=$?
+elif [ -w "/sys/power/state" ] ; then
+	# Use the raw kernel sysfs interface
+	echo "disk" > /sys/power/state
+	RET=$?
 else
 	unsupported
 fi
diff -urN hal-0.5.10/tools/linux/hal-system-power-set-power-save-linux hal-0.5.10.new/tools/linux/hal-system-power-set-power-save-linux
--- hal-0.5.10/tools/linux/hal-system-power-set-power-save-linux	2007-07-26 20:00:28.000000000 +0200
+++ hal-0.5.10.new/tools/linux/hal-system-power-set-power-save-linux	2007-10-14 00:17:08.000000000 +0200
@@ -17,7 +17,14 @@
 		/usr/sbin/pm-powersave false
 		RET=$?
 	fi
-
+elif [ -x "/usr/bin/powersave" ] ; then
+	if [ $value = "true" ]; then
+		/usr/bin/powersave -e Powersave
+		RET=$?
+	elif [ $value = "false" ]; then
+		/usr/bin/powersave -e Performance
+		RET=$?
+	fi
 else
 	# cannot set proc stuff here, so error out
 	unsupported
diff -urN hal-0.5.10/tools/linux/hal-system-power-suspend-linux hal-0.5.10.new/tools/linux/hal-system-power-suspend-linux
--- hal-0.5.10/tools/linux/hal-system-power-suspend-linux	2007-07-26 20:00:28.000000000 +0200
+++ hal-0.5.10.new/tools/linux/hal-system-power-suspend-linux	2007-10-14 00:23:36.000000000 +0200
@@ -39,6 +39,19 @@
 if [ -x "/usr/sbin/pm-suspend" ] ; then
 	/usr/sbin/pm-suspend $QUIRKS
 	RET=$?
+elif [ -x "/usr/sbin/s2ram" ] ; then
+	/usr/sbin/s2ram
+	RET=$?
+elif [ -x "/usr/bin/powersave" ] ; then
+	dbus-send --system --dest=com.novell.powersave --print-reply /com/novell/powersave com.novell.powersave.action.SuspendToRam
+	RET=$?
+elif [ -x "/usr/sbin/pmi" ] ; then
+	/usr/sbin/pmi action suspend force
+	RET=$?
+elif [ -w "/sys/power/state" ] ; then
+	# Use the raw kernel sysfs interface
+	echo "mem" > /sys/power/state
+	RET=$?
 else
 	# TODO: add support
 	unsupported
