--- hal-0.5.2/tools/device-manager/DeviceManager.py.orig	2005-04-27 21:15:37.000000000 +0200
+++ hal-0.5.2/tools/device-manager/DeviceManager.py	2005-05-19 16:40:29.000000000 +0200
@@ -261,13 +261,16 @@
         for name in device_names:
             device_dbus_obj = self.bus.get_object("org.freedesktop.Hal" ,name)
             properties = device_dbus_obj.GetAllProperties(dbus_interface="org.freedesktop.Hal.Device")
-            try:
-                parent_name = properties["info.parent"]
-            except KeyError:
-                # no parent, must be parent of virtual_root
-                parent_name = "/"
-            device = Device(name, parent_name, properties)
-            self.device_list.append(device)
+            if properties == None:
+                print name + " has no properties"
+            else:
+                try:
+                    parent_name = properties["info.parent"]
+                except KeyError:
+                    # no parent, must be parent of virtual_root
+                    parent_name = "/"
+                device = Device(name, parent_name, properties)
+                self.device_list.append(device)
 
         # set parent_device and children for each Device object
         for device in self.device_list:
