--- hal-0.5.10/configure.in.orig	2007-10-31 20:02:08.847844000 +0100
+++ hal-0.5.10/configure.in	2007-10-31 20:20:27.462450588 +0100
@@ -242,8 +242,8 @@
 if test "x$use_parted" = "xyes" ; then
   USE_PARTED=yes
   AC_DEFINE(USE_PARTED,1,[Whether libparted is to be used])
-  AC_CHECK_LIB(uuid, uuid_generate, [], AC_MSG_ERROR([*** uuid library (libuuid) not found]))
-  AC_CHECK_LIB(dl, dlopen,          [], AC_MSG_ERROR([*** dl library (libdl) not found]))
+  AC_CHECK_LIB(uuid, uuid_generate, [:], AC_MSG_ERROR([*** uuid library (libuuid) not found]))
+  AC_CHECK_LIB(dl, dlopen,          [:], AC_MSG_ERROR([*** dl library (libdl) not found]))
 
   AC_MSG_CHECKING([for libparted == 1.7.1 or >= 1.8.0])
   if ! $PKG_CONFIG --atleast-version 1.8.0 libparted; then
@@ -417,6 +417,7 @@
 	AC_LANG_POP([C++])
 	if test "$LIB_SMBIOS" = "yes" ; then
    		AC_MSG_CHECKING([for libsmbios >= 0.13.4])
+		save_LIBS="$LIBS"
    		LIBS="$LIBS -lsmbios"
    		AC_RUN_IFELSE([AC_LANG_SOURCE([[
    		#include <smbios/version.h>
@@ -436,6 +437,7 @@
 			return 1;
    		}]])], [USE_SMBIOS=yes; AC_MSG_RESULT(yes); AM_CONDITIONAL(HAVE_SMBIOS,true)],
        		[USE_SMBIOS=no; AC_MSG_RESULT(failed); AM_CONDITIONAL(HAVE_SMBIOS,false)])
+		LIBS="$save_LIBS"
 	else
   		USE_SMBIOS=no
   		AM_CONDITIONAL(HAVE_SMBIOS, [false])
