Index: fdi/90defaultpolicy/storage-policy.fdi
===================================================================
RCS file: /cvs/hal/hal/fdi/90defaultpolicy/storage-policy.fdi,v
retrieving revision 1.7
diff -u -r1.7 storage-policy.fdi
--- fdi/90defaultpolicy/storage-policy.fdi	25 Oct 2004 16:32:10 -0000	1.7
+++ fdi/90defaultpolicy/storage-policy.fdi	26 Oct 2004 18:20:14 -0000
@@ -171,12 +171,22 @@
 	    <match key="volume.partition.msdos_part_table_type" int="0x0e">
 	      <merge key="volume.policy.should_mount" type="bool">true</merge>
 	    </match>
-	  </match>
-
-	  
+	  </match>	  
 	</match>
       </match>
     </match>
+    
+  </device>
+
+  <!-- Dont want to mount non-hotpluggable fixed disks since ideraid
+       detection isnt complete as hald wrongly detects e.g. partitions
+       from some IDE RAID controllers -->
+  <device>
+    <match key="storage.hotpluggable" bool="false">
+      <match key="storage.removable" bool="false">
+	<merge key="storage.policy.should_mount" type="bool">false</merge>
+      </match>
+    </match>
   </device>
 
 </deviceinfo>
