--- hal-0.4.7/configure.in.orig	2005-01-25 03:41:42.000000000 +0100
+++ hal-0.4.7/configure.in	2005-03-07 23:04:53.839924760 +0100
@@ -67,7 +67,7 @@
 if test "x$enable_fstab_sync" = "xyes" ; then
    # fstab-sync needs popt, so check for it here
    AC_CHECK_HEADERS(popt.h,, [AC_MSG_ERROR([You need popt to build fstab-sync])])
-   AC_CHECK_LIB(popt, poptGetContext,, [AC_MSG_ERROR([You need popt to build fstab-sync])])
+   AC_CHECK_LIB(popt, poptGetContext,[:], [AC_MSG_ERROR([You need popt to build fstab-sync])])
 
    AC_DEFINE(FSTAB_SYNC_ENABLED,1,[Whether fstab-sync callout should be installed])
 fi
