diff -aurN hal-0.4.1.orig/tools/device-manager/DeviceManager.py hal-0.4.1/tools/device-manager/DeviceManager.py
--- hal-0.4.1.orig/tools/device-manager/DeviceManager.py	2004-11-08 23:33:39.000000000 +0100
+++ hal-0.4.1/tools/device-manager/DeviceManager.py	2004-11-27 01:00:42.976801072 +0100
@@ -265,13 +265,16 @@
             device_dbus_obj = self.hal_service.get_object(name,
                                                      "org.freedesktop.Hal.Device")
             properties = device_dbus_obj.GetAllProperties()
-            try:
+            if properties == None:
+               print name + " has no properties"
+            else:
+             try:
                 parent_name = properties["info.parent"]
-            except KeyError:
+             except:
                 # no parent, must be parent of virtual_root
                 parent_name = "/"
-            device = Device(name, parent_name, properties)
-            self.device_list.append(device)
+             device = Device(name, parent_name, properties)
+             self.device_list.append(device)
 
         # set parent_device and children for each Device object
         for device in self.device_list:
