diff -aurN hal-0.5.6.orig/hald/Makefile.am hal-0.5.6/hald/Makefile.am
--- hal-0.5.6.orig/hald/Makefile.am	2005-12-08 19:23:23.000000000 +0100
+++ hal-0.5.6/hald/Makefile.am	2006-02-18 19:09:56.481409312 +0100
@@ -9,7 +9,7 @@
 	-DPACKAGE_BIN_DIR=\""$(bindir)"\" \
 	-DPACKAGE_LOCALE_DIR=\""$(prefix)/$(DATADIRNAME)/locale"\" \
 	-DPACKAGE_LOCALSTATEDIR=\""$(localstatedir)"\" \
-	-DPACKAGE_SCRIPT_DIR=\""$(datadir)/hal/scripts"\" \
+	-DPACKAGE_SCRIPT_DIR=\""$(libdir)/hal/scripts"\" \
 	-I$(top_srcdir) \
 	@PACKAGE_CFLAGS@
 
diff -aurN hal-0.5.6.orig/tools/Makefile.am hal-0.5.6/tools/Makefile.am
--- hal-0.5.6.orig/tools/Makefile.am	2005-12-21 18:08:08.000000000 +0100
+++ hal-0.5.6/tools/Makefile.am	2006-02-18 19:09:56.832355960 +0100
@@ -43,7 +43,7 @@
 endif
 ## FSTAB_SYNC_ENABLED end
 
-scriptdir = $(datadir)/hal/scripts
+scriptdir = $(libdir)/hal/scripts
 
 script_SCRIPTS =				\
 	hal-luks-setup hal-luks-remove		\
