diff -aurN hal-0.5.4.orig/fdi/policy/10osvendor/10-storage-policy.fdi hal-0.5.4/fdi/policy/10osvendor/10-storage-policy.fdi
--- hal-0.5.4.orig/fdi/policy/10osvendor/10-storage-policy.fdi	2005-05-23 16:06:27.000000000 +0200
+++ hal-0.5.4/fdi/policy/10osvendor/10-storage-policy.fdi	2005-09-04 15:10:20.000000000 +0200
@@ -8,9 +8,9 @@
       <merge key="storage.policy.default.mount_root" type="string">/media</merge>
       <merge key="storage.policy.default.use_managed_keyword" type="bool">true</merge>
       <merge key="storage.policy.default.managed_keyword.primary" type="string">managed</merge>
-      <merge key="storage.policy.default.managed_keyword.secondary" type="string">kudzu</merge>
+      <merge key="storage.policy.default.managed_keyword.secondary" type="string"></merge>
       <merge key="storage.policy.default.mount_option.noauto" type="bool">true</merge>
-      <merge key="storage.policy.default.mount_option.pamconsole" type="bool">true</merge>
+      <merge key="storage.policy.default.mount_option.user" type="bool">true</merge>
       <merge key="storage.policy.default.mount_option.exec" type="bool">true</merge>
     </match>
   </device>
@@ -44,6 +44,19 @@
       </match>
     </match>
 
+    <!-- Use UTF-8 charset and the quiet option for vfat -->
+    <match key="volume.fstype" string="vfat">
+      <merge key="volume.policy.mount_option.iocharset=utf8" type="bool">true</merge>
+      <merge key="volume.policy.mount_option.quiet" type="bool">true</merge>
+    </match>
+    <!-- cdrom filesystems should be mounted with the utf8 option -->
+    <match key="volume.fstype" string="iso9660">
+      <merge key="volume.policy.mount_option.utf8" type="bool">true</merge>
+    </match>
+    <match key="volume.fstype" string="udf">
+      <merge key="volume.policy.mount_option.utf8" type="bool">true</merge>
+    </match>
+
     <!-- Handle drives with non-partitioned media  -->
     <match key="storage.no_partitions_hint" bool="true">
       <!-- optical drives -->
@@ -51,25 +64,25 @@
 	<merge key="storage.policy.mount_filesystem" type="string">auto</merge>
 	<merge key="storage.policy.desired_mount_point" type="string">cdrom</merge>
 	<match key="storage.cdrom.cdr" bool="true">
-	  <merge key="storage.policy.desired_mount_point" type="string">cdrecorder</merge>
+	  <merge key="storage.policy.desired_mount_point" type="string">cdwriter</merge>
 	</match>
 	<match key="storage.cdrom.cdrw" bool="true">
-	  <merge key="storage.policy.desired_mount_point" type="string">cdrecorder</merge>
+	  <merge key="storage.policy.desired_mount_point" type="string">cdwriter</merge>
 	</match>
 	<match key="storage.cdrom.dvdplusr" bool="true">
-	  <merge key="storage.policy.desired_mount_point" type="string">cdrecorder</merge>
+	  <merge key="storage.policy.desired_mount_point" type="string">dvdwriter</merge>
 	</match>
 	<match key="storage.cdrom.dvdplusrw" bool="true">
-	  <merge key="storage.policy.desired_mount_point" type="string">cdrecorder</merge>
+	  <merge key="storage.policy.desired_mount_point" type="string">dvdwriter</merge>
 	</match>
 	<match key="storage.cdrom.dvdram" bool="true">
-	  <merge key="storage.policy.desired_mount_point" type="string">cdrecorder</merge>
+	  <merge key="storage.policy.desired_mount_point" type="string">dvdwriter</merge>
 	</match>
 	<match key="storage.cdrom.dvdr" bool="true">
-	  <merge key="storage.policy.desired_mount_point" type="string">cdrecorder</merge>
+	  <merge key="storage.policy.desired_mount_point" type="string">dvdwriter</merge>
 	</match>
 	<match key="storage.cdrom.dvdrw" bool="true">
-	  <merge key="storage.policy.desired_mount_point" type="string">cdrecorder</merge>
+	  <merge key="storage.policy.desired_mount_point" type="string">dvdwriter</merge>
 	</match>
       </match>
 
@@ -96,6 +109,9 @@
     <match key="block.is_volume" bool="true">
       <match key="volume.fsusage" string="filesystem">
 	<!-- skip for drives with the no partitions hint (they are handled above) -->
+	<match key="@block.storage_device:storage.no_partitions_hint" bool="true">
+	    <merge key="volume.policy.desired_mount_point" type="copy_property">@block.storage_device:storage.policy.desired_mount_point</merge>
+	</match>
 	<match key="@block.storage_device:storage.no_partitions_hint" bool="false">
 
 	  <merge key="volume.policy.should_mount" type="bool">true</merge>
@@ -195,5 +200,14 @@
       <append key="info.addons" type="strlist">hald-addon-storage</append>
     </match>
   </device>
+
+  <!-- Always eject USB storage devices to properly power them off -->
+  <device>
+    <match key="info.category" string="storage">
+      <match key="storage.bus" string="usb">
+        <merge key="storage.requires_eject" type="bool">true</merge>
+      </match>
+    </match>
+  </device>
   
 </deviceinfo>
