diff -aurN hal-0.5.7.1.orig/hald/linux2/osspec.c hal-0.5.7.1/hald/linux2/osspec.c
--- hal-0.5.7.1.orig/hald/linux2/osspec.c	2006-02-13 17:42:32.000000000 +0100
+++ hal-0.5.7.1/hald/linux2/osspec.c	2006-07-27 20:53:52.859809168 +0200
@@ -476,6 +476,8 @@
 		can_hibernate = TRUE;
 	if (access ("/proc/suspend2", F_OK) == 0)
 		can_hibernate = TRUE;
+	if (access ("/sys/power/suspend2", F_OK) == 0)
+		can_hibernate = TRUE;
 out:
 	hal_device_property_set_bool (d, "power_management.can_suspend_to_ram", can_suspend);
 	hal_device_property_set_bool (d, "power_management.can_suspend_to_disk", can_hibernate);
