--- hal-0.5.9/configure.in.orig	2007-04-03 06:36:44.000000000 +0200
+++ hal-0.5.9/configure.in	2007-05-24 19:13:20.983232115 +0200
@@ -191,8 +191,8 @@
 if test "x$use_parted" = "xyes" ; then
    USE_PARTED=yes
    AC_DEFINE(USE_PARTED,1,[Whether libparted is to be used])
-   AC_CHECK_LIB(uuid, uuid_generate, [], AC_MSG_ERROR([*** uuid library (libuuid) not found]))
-   AC_CHECK_LIB(dl, dlopen,          [], AC_MSG_ERROR([*** dl library (libdl) not found]))
+   AC_CHECK_LIB(uuid, uuid_generate, [:], AC_MSG_ERROR([*** uuid library (libuuid) not found]))
+   AC_CHECK_LIB(dl, dlopen,          [:], AC_MSG_ERROR([*** dl library (libdl) not found]))
    AC_MSG_CHECKING([for libparted == 1.7.1, 1.8.0, 1.8.1, 1.8.2, 1.8.6])
    LDFLAGS=-lparted
    AC_TRY_RUN(
@@ -211,7 +211,8 @@
 	    (major == 1 && minor == 8 && micro == 0) ||
 	    (major == 1 && minor == 8 && micro == 1) ||
 	    (major == 1 && minor == 8 && micro == 2) ||
-	    (major == 1 && minor == 8 && micro == 6))
+	    (major == 1 && minor == 8 && micro == 6) ||
+	    (major == 1 && minor == 8 && micro == 7))
 	   return 0;
 
 	return 1;
