diff -urN hal-0.5.8.1.org/tools/linux/hal-system-power-hibernate-linux hal-0.5.8.1/tools/linux/hal-system-power-hibernate-linux
--- hal-0.5.8.1.org/tools/linux/hal-system-power-hibernate-linux	2006-09-19 22:23:25.000000000 +0200
+++ hal-0.5.8.1/tools/linux/hal-system-power-hibernate-linux	2006-10-31 21:21:28.542805500 +0100
@@ -10,56 +10,29 @@
 	exit 1
 }
 
-#SuSE and ALTLinux only support powersave
-if [ -f /etc/altlinux-release ] || [ -f "/etc/SuSE-release" ] ; then
-	if [ -x /usr/bin/powersave ] ; then
-	        $POWERSAVED_SUSPEND2DISK
-		RET=$?
-	else
-		unsupported
-	fi
-
-#Mandriva support suspend-scripts 
-elif [ -f /etc/mandriva-release ] ; then 
-    if [ -x /usr/sbin/pmsuspend ] ; then 
-	/usr/sbin/pmsuspend disk 
-	RET=$? 
-    else 
-	unsupported 
-    fi 
-
-#RedHat/Fedora only support pm-utils
-elif [ -f /etc/redhat-release ] || [ -f /etc/fedora-release ] ; then
-	if [ -x /usr/sbin/pm-hibernate ] ; then
-		/usr/sbin/pm-hibernate
-		RET=$?
-	else
-		unsupported
-	fi
-
-#Other distros just need to have *any* tools installed
+if [ -x /usr/sbin/s2disk ] ; then
+    /usr/sbin/s2disk
+    RET=$?
+elif [ -x "/usr/sbin/hibernate" ] ; then
+    # Suspend2 tools installed
+    /usr/sbin/hibernate --force
+    RET=$?
+elif [ -x /usr/bin/powersave ] ; then
+    $POWERSAVED_SUSPEND2DISK
+    RET=$?
+elif [ -x /usr/sbin/pmsuspend ] ; then 
+    /usr/sbin/pmsuspend disk 
+    RET=$? 
+elif [ -x /usr/sbin/pm-hibernate ] ; then
+    /usr/sbin/pm-hibernate
+    RET=$?
+elif [ -w "/sys/power/state" ] ; then
+    # Use the raw kernel sysfs interface
+    echo "disk" > /sys/power/state
+    RET=$?
 else
-	if [ -x "/usr/bin/powersave" ] ; then
-	        $POWERSAVED_SUSPEND2DISK
-		RET=$?
-	elif [ -x "/usr/sbin/pmi" ] ; then
-		/usr/sbin/pmi action hibernate force
-		RET=$?
-	elif [ -x "/usr/sbin/pm-hibernate" ] ; then
-		/usr/sbin/pm-hibernate
-		RET=$?
-	elif [ -x "/usr/sbin/hibernate" ] ; then
-		# Suspend2 tools installed
-		/usr/sbin/hibernate --force
-		RET=$?
-	elif [ -w "/sys/power/state" ] ; then
-		# Use the raw kernel sysfs interface
-		echo "disk" > /sys/power/state
-		RET=$?
-	else
-		unsupported
-		fi
-	fi
+    unsupported
+fi
 
 #Refresh devices as a resume can do funny things
 for type in button battery ac_adapter
diff -urN hal-0.5.8.1.org/tools/linux/hal-system-power-set-power-save-linux hal-0.5.8.1/tools/linux/hal-system-power-set-power-save-linux
--- hal-0.5.8.1.org/tools/linux/hal-system-power-set-power-save-linux	2006-09-19 22:23:25.000000000 +0200
+++ hal-0.5.8.1/tools/linux/hal-system-power-set-power-save-linux	2006-10-31 21:18:22.463176250 +0100
@@ -7,38 +7,25 @@
 	exit 1
 }
 
-#SuSE and ALTLinux only support powersave
-if [ -f /etc/altlinux-release ] || [ -f /etc/SuSE-release ] ; then
-	if [ -x "/usr/bin/powersave" ] ; then
-		if [ $value = "true" ]; then
-			/usr/bin/powersave -e Powersave
-			RET=$?
-		elif [ $value = "false" ]; then
-			/usr/bin/powersave -e Performance
-			RET=$?
-		fi
-	else
-		unsupported
+if [ -x "/usr/bin/powersave" ] ; then
+	if [ $value = "true" ]; then
+		/usr/bin/powersave -e Powersave
+		RET=$?
+	elif [ $value = "false" ]; then
+		/usr/bin/powersave -e Performance
+		RET=$?
 	fi
-
-#RedHat/Fedora only support pm-utils
-elif [ -f /etc/redhat-release ] || [ -f /etc/fedora-release ] ; then
-	if [ -x "/usr/sbin/pm-powersave" ] ; then
-		if [ $value = "true" ]; then
-			/usr/sbin/pm-powersave true
-			RET=$?
-		elif [ $value = "false" ]; then
-			/usr/sbin/pm-powersave false
-			RET=$?
-		fi
-	else
-		unsupported
+elif [ -x "/usr/sbin/pm-powersave" ] ; then
+	if [ $value = "true" ]; then
+		/usr/sbin/pm-powersave true
+		RET=$?
+	elif [ $value = "false" ]; then
+		/usr/sbin/pm-powersave false
+		RET=$?
 	fi
-
 else
-	# cannot set proc stuff here, so error out
 	unsupported
-	fi 
+fi
 
 exit $RET
 
diff -urN hal-0.5.8.1.org/tools/linux/hal-system-power-suspend-linux hal-0.5.8.1/tools/linux/hal-system-power-suspend-linux
--- hal-0.5.8.1.org/tools/linux/hal-system-power-suspend-linux	2006-09-19 22:23:25.000000000 +0200
+++ hal-0.5.8.1/tools/linux/hal-system-power-suspend-linux	2006-10-31 21:20:50.544430750 +0100
@@ -28,73 +28,26 @@
 	exit 0
 fi
 
-#SuSE and ALTLinux only support powersave
-if [ -f "/etc/altlinux-release" ] || [ -f "/etc/SuSE-release" ] ; then
-	if [ -x /usr/bin/powersave ] ; then
-	        $POWERSAVED_SUSPEND2RAM
-		RET=$?
-	else
-		# TODO: add support
-		unsupported
-	fi
-
-#Mandriva supports suspend-scripts 
-elif [ -f "/etc/mandriva-release" ] ; then 
-    # TODO: fix pmsuspend to take a --wakeup-alarm argument 
-    if [ $seconds_to_sleep != "0" ] ; then 
-	alarm_not_supported 
-    fi 
-    
-    if [ -x "/usr/sbin/pmsuspend" ] ; then 
-	/usr/sbin/pmsuspend memory
-	RET=$? 
-    else 
-        # TODO: add support 
-	unsupported 
-    fi 
-
-#RedHat/Fedora only support pm-utils
-elif [ -f "/etc/redhat-release" ] || [ -f "/etc/fedora-release" ] ; then
-	# TODO: fix pm-suspend to take a --wakeup-alarm argument
-	if [ $seconds_to_sleep != "0" ] ; then
-		alarm_not_supported
-	fi
-	# TODO: fixup pm-suspend to define erroc code (see alarm above) and throw
-	#	   the appropriate exception
-	if [ -x "/usr/sbin/pm-suspend" ] ; then
-		/usr/sbin/pm-suspend
-		RET=$?
-	else
-		# TODO: add support
-		unsupported
-	fi
-
-#FreeBSD uses zzz to suspend for both ACPI and APM
-elif [ "x`uname -s`" = "xFreeBSD" ] ; then
-	if [ -x /usr/sbin/zzz ] ; then
-		/usr/sbin/zzz
-		RET=$?
-	else
-		unsupported
-	fi
-
-#Other distros just need to have *any* tools installed
+if [ -x /usr/sbin/s2ram ]; then
+	/usr/sbin/s2ram
+	RET=$?
+elif [ -x /usr/bin/powersave ] ; then
+        $POWERSAVED_SUSPEND2RAM
+	RET=$?
+elif [ -x "/usr/sbin/pm-suspend" ] ; then
+	/usr/sbin/pm-suspend
+	RET=$?
+elif [ -x "/usr/sbin/pmi" ] ; then
+        /usr/sbin/pmi action suspend force
+	RET=$?
+elif [ -w "/sys/power/state" ] ; then
+        # Use the raw kernel sysfs interface
+        echo "mem" > /sys/power/state
+	RET=$?
 else
-	if [ -x "/usr/bin/powersave" ] ; then
-	    $POWERSAVED_SUSPEND2RAM
-	    RET=$?
-	elif [ -x "/usr/sbin/pmi" ] ; then
-	    /usr/sbin/pmi action suspend force
-	    RET=$?
-	elif [ -w "/sys/power/state" ] ; then
-	    # Use the raw kernel sysfs interface
-	    echo "mem" > /sys/power/state
-	    RET=$?
-	else
-	    # TODO: add other scripts support
-	    unsupported
-	    fi
-	fi
+        # TODO: add other scripts support
+        unsupported
+fi
 
 #Refresh devices as a resume can do funny things
 for type in button battery ac_adapter
